### PR TITLE
Fix logging and new name for allowlist

### PR DESCRIPTION
**Pull request by lmazuel**
Fix https://github.com/Azure/azure-sdk-for-python/issues/26331

Note: It's incredibly more difficult to hide an alias to a class property than it seems, as there is no class level `getattr`. But the metaclass option works fine, and is not read by Sphinx:

image

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/policies/_universal.py [hunk 1]
```diff
@@ -351,15 +351,27 @@ def on_response(self, request, response):
...
             _LOGGER.debug("Failed to log response: %s", repr(err))
 
+class _HiddenClassProperties(type):
+    # Backward compatible for DEFAULT_HEADERS_WHITELIST but no documentation needed
+    # https://github.com/Azure/azure-sdk-for-python/issues/26331
 
-class HttpLoggingPolicy(SansIOHTTPPolicy):
+    @property
+    def DEFAULT_HEADERS_WHITELIST(cls):
+        return cls.DEFAULT_HEADERS_ALLOWLIST
```
Comment (annatisch) [inline sdk/core/azure-core/azure/core/pipeline/policies/_universal.py:360]: Do we want to issue a deprecation warning if this is used? Or will that be too noisy for the time being?

File: sdk/core/azure-core/azure/core/pipeline/policies/_universal.py [hunk 2]
```diff
@@ -351,15 +351,27 @@ def on_response(self, request, response):
...
+
+    @DEFAULT_HEADERS_WHITELIST.setter
+    def DEFAULT_HEADERS_WHITELIST(self, value):
+        self.DEFAULT_HEADERS_ALLOWLIST = value
```
Comment (annatisch) [inline sdk/core/azure-core/azure/core/pipeline/policies/_universal.py:364]: This only works to set on an instance(`self`) not the class type (`cls`)?

Comment (annatisch) [reply sdk/core/azure-core/azure/core/pipeline/policies/_universal.py:364]: Ignore that - I see in your tests you're testing setting to the class type - so looks like it's good.

File: sdk/core/azure-mgmt-core/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,17 @@
 # Release History
 
+## 1.3.3 (Unreleased)
```
Comment (annatisch) [inline sdk/core/azure-mgmt-core/CHANGELOG.md:3]: This change is a patch, not a minor?

**Review by annatisch — APPROVED** [review]

File: sdk/core/azure-core/azure/core/pipeline/policies/_universal.py (hunk 2 above)
Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/_universal.py:364]: Ha yeah, used `self` by habit, should have called it `cls`

File: sdk/core/azure-mgmt-core/CHANGELOG.md (hunk 3 above)
Comment (lmazuel) [reply sdk/core/azure-mgmt-core/CHANGELOG.md:3]: Is that really a new feature?

File: sdk/core/azure-core/azure/core/pipeline/policies/_universal.py (hunk 1 above)
Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/_universal.py:360]: I think it's too noisy, we don't have any reasons to remove it for now. Maybe if we feel it becomes a maintenance burden, we can add it later when it was not documented for months

**Comment by lmazuel** [comment]
The few remaining red CI are unrelated, merging

**Pull request merged**